pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -93,7 +93,15 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 		if rs.PodSpec.Containers[idx].ReadinessProbe.SuccessThreshold == 0 {
 			rs.PodSpec.Containers[idx].ReadinessProbe.SuccessThreshold = 1
 		}
+		if rs.PodSpec.Containers[idx].ReadinessProbe.PeriodSeconds != 0 {
```

Comment:
Is this check necessary here?

Reply:
@taragu yes i think its required 
Because consider in service file not given any value for `PeriodSeconds, FailureThreshold and TimeoutSeconds` 
than if we don't add this condition what will happen is `FailureThreshold and TimeoutSeconds` will get default values and because of that [Knative's special probe](https://github.com/knative/serving/blob/master/pkg/apis/serving/k8s_validation.go#L463) scenario will fail

Reply:
This sounds familiar to what was discussed here https://github.com/knative/serving/pull/4780#issuecomment-514719066.

Adding @dgerd to chime in here.